lib/ansible/modules/source_control/git.py

```diff
@@ -186,6 +186,33 @@
               to be installed. The commit MUST be signed and the public key MUST
               be trusted in the GPG trustdb.
 
+    archive:
+        required: false
+        default: "no"
+        choices: [ "yes", "no" ]
+        version_added: "2.4"
+        description:
+            - if C(yes), creates an archive of the specified format containing
+              the tree structure for the source tree.
+
+    archive_format:
+        required: false
+        default: "tar"
+        choices: [ "zip", "tar", "tar.gz" ]
+        version_added: "2.4"
+        description:
+            - if specified, then creates archive in specified format.
+              Default is "tar"
+
+    archive_file:
+        required: false
+        default: null
+        version_added: "2.4"
+        description:
+            - Required parameter if archive is specified. If not specified
```

Comment:
This doesn't make sense. You say it's required, and then say it will default to something if it's not set.